### ISSUE TITLE
rm -f returns "You cannot remove a running container. Stop the container before attempting removal or use -f"

### ISSUE THREAD

**Issue by hickeng** [vmware/vic#6263]
Running the following trivial test against bart.eng.vmware.com

```
vmware@vmware-virtual-machine:~/vic/bin$ while true; do
> ((i++));
> echo $i;
> rm -f /tmp/test-done-{1,2,3};
>
> ( docker run -dit -v vol1:/mnt/test --name=helloworld1 --net=external alpine && docker rename helloworld1 goodbyeworld1 && docker rm -f -v goodbyeworld1 && touch /tmp/test-done-1 ) &
> ( docker run -dit -v vol2:/mnt/test --name=helloworld2 --net=external alpine && docker rename helloworld2 goodbyeworld2 && docker rm -f -v goodbyeworld2 && touch /tmp/test-done-2 ) &
> ( docker run -dit -v vol3:/mnt/test --name=helloworld3 --net=external alpine && docker rename helloworld3 goodbyeworld3 && docker rm -f -v goodbyeworld3 && touch /tmp/test-done-3 ) &
>
> until test -f /tmp/test-done-1; do sleep 1; done;
> until test -f /tmp/test-done-2; do sleep 1; done;
> until test -f /tmp/test-done-3; do sleep 1; done;
>
> sleep 2;
> done
2
[1] 4397
[2] 4398
[3] 4399
3ab9ee2764339048a6d832dd0a536803c4009385020bf90d147348968828e201
4de55e22850b6004133f81b5ea45f53500c68173edec741c47057759ae2ada36
dd1a37c7e15db337e5bdd39e4b8b3d33892536b998f423018091cd80b1606845
Error response from daemon: You cannot remove a running container. Stop the container before attempting removal or use -f
[2]-  Exit 1                  ( docker --tls run -dit -v vol2:/mnt/test --name=helloworld2 --net=external alpine && docker --tls rename helloworld2 goodbyeworld2 && docker --tls rm -f -v goodbyeworld2 && touch /tmp/test-done-2 )
goodbyeworld1
[1]-  Done                    ( docker --tls run -dit -v vol1:/mnt/test --name=helloworld1 --net=external alpine && docker --tls rename helloworld1 goodbyeworld1 && docker --tls rm -f -v goodbyeworld1 && touch /tmp/test-done-1 )
goodbyeworld3
[3]+  Done                    ( docker --tls run -dit -v vol3:/mnt/test --name=helloworld3 --net=external alpine && docker --tls rename helloworld3 goodbyeworld3 && docker --tls rm -f -v goodbyeworld3 && touch /tmp/test-done-3 )
```

[container-logs (38).zip](https://github.com/vmware/vic/files/1290448/container-logs.38.zip)

**Comment by jzt** [comment]
Seen in [5-4-High-Availability.zip](https://github.com/vmware/vic/files/1350118/5-4-High-Availability.zip)

**Comment by hickeng** [comment]
Estimate is predicated on this being addressed by #6252 and is for confirmation.
If this is not addressed by that work then it needs re-estimating.

**Comment by matthewavery** [comment]
Seen during a nightly VC 6.5 5-4-High-Availability run on 11/4/2017.
This is VIC Engine Build 14394

[5-4-High-Availability.zip](https://github.com/vmware/vic/files/1446884/5-4-High-Availability.zip)

**Comment by matthewavery** [comment]
linking #6252 and #6672 these also encompass work on this reported bug. I am leaving all logs on this issue as it lines up best with the nightly failures we are seeing.

**Comment by matthewavery** [comment]
Seen during a nightly VC 6.0 5-4-High-Availability failure on 11/7/2017.
This is VIC Engine Build 14432

[5-4-High-Availability.zip](https://github.com/vmware/vic/files/1454630/5-4-High-Availability.zip)

**Comment by sgairo** [comment]
marking as high priority for 1.3 release.

**Comment by sflxn** [comment]
I believe #6252 is the same issue.  There is more debugging done there.  I'm going to focus work in that issue but leave this issue open as it has all the CI logs.

**Comment by sgairo** [comment]
This is seen in the 6.0 11/19 nightly report: 
[5-4-High-Availability.zip](https://github.com/vmware/vic/files/1488221/5-4-High-Availability.zip)

`ERROR Handler for DELETE /v1.25/containers/0301288c4c4fc33ac8aa8fcd169c9819e8596f8bf223531dfcf53be252ce2e2e returned error: You cannot remove a running container. Stop the container before attempting removal or use -f`  which then caused:

`ERROR Handler for DELETE /v1.25/images/busybox returned error: Failed to remove image "busybox": image 789355058656ec2ca0a856283e2597852ccd444548c9ef4fcbf8c9c24a73e169 in use by 0301288c4c4fc33ac8aa8fcd169c9819e8596f8bf223531dfcf53be252ce2e2e`

**Comment by mhagen-vmware** [comment]
should have closed with #6786
